generators/server/index.js

```diff
@@ -250,6 +246,25 @@ module.exports = JhipsterServerGenerator.extend({
                 },
                 {
                     when: function (response) {
+                        return applicationType == 'monolith' && (response.authenticationType == 'session' || response.authenticationType == 'jwt');
+                    },
+                    type: 'list',
+                    name: 'enableSocialSignIn',
```

Comment:
you have to change the current tests 
https://github.com/jhipster/generator-jhipster/blob/master/test/test-creation.js#L676-L692